kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/clients/records/StrimziTestClientConsumerRecord.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.systemtests.clients.records;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map.Entry;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * The type Strimzi Test Client consumer record.
+ */
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class StrimziTestClientConsumerRecord extends ConsumerRecord {
+
+    /**
+     * Instantiates a new Client consumer record.
+     *
+     * @param headers the headers
+     * @param topic the topic
+     * @param key the key
+     * @param payload the payload
+     * @param partition the partition
+     * @param offset the offset
+     */
+    @JsonCreator
+    public StrimziTestClientConsumerRecord(@JsonProperty("headers") List<Entry<String, String>> headers,
+                                           @JsonProperty("topic") String topic,
+                                           @JsonProperty("key") String key,
+                                           @JsonProperty("payload") String payload,
+                                           @JsonProperty("partition") int partition,
+                                           @JsonProperty("offset") long offset) {
+        this.recordHeaders = new HashMap<>();
+        if (headers != null) {
+            headers.forEach(h -> recordHeaders.put(h.getKey(), h.getValue()));
```

Comment:
```suggestion
            recordHeaders.entrySet().addAll(headers);
```

Reply:
Tested but it does not work:
```
Caused by: java.lang.UnsupportedOperationException
	at java.base/java.util.AbstractCollection.add(AbstractCollection.java:253)
	at java.base/java.util.AbstractCollection.addAll(AbstractCollection.java:338)
	at io.kroxylicious.systemtests.clients.records.StrimziTestClientConsumerRecord.<init>(StrimziTestClientConsumerRecord.java:42)
	at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:502)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:486)
	at com.fasterxml.jackson.databind.introspect.AnnotatedConstructor.call(AnnotatedConstructor.java:126)
	at com.fasterxml.jackson.databind.deser.std.StdValueInstantiator.createFromObjectWith(StdValueInstantiator.java:291)
```

Reply:
Looking into the java library class I found this:

```
public boolean addAll(Collection<? extends E> c) {
        boolean modified = false;
        for (E e : c)
            if (add(e))
                modified = true;
        return modified;
    }
```
pointing to `add(e)` that is not supported:
```
public boolean add(E e) {
        throw new UnsupportedOperationException();
    }
```